internal/ctxutil/ctx.go

```diff
@@ -0,0 +1,38 @@
+package ctxutil
+
+import (
+	"context"
+	"github.com/TykTechnologies/tyk/internal/proxy"
+	"net/http"
+)
+
+const (
+	upstreamAuth = "upstream-auth"
+)
+
+func SetContext(r *http.Request, ctx context.Context) {
```

Comment:
Since the object modified is from the http namespace, this should likely be httputil.SetContext or similar